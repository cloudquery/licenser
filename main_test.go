package main

import (
	"os"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestCheck(t *testing.T) {
	cmd := rootCmd()
	cmd.SetArgs([]string{"check", "."})
	err := cmd.Execute()
	require.NoError(t, err)
}

func TestReport(t *testing.T) {
	cmd := rootCmd()
	cmd.SetArgs([]string{"report", "."})
	err := cmd.Execute()
	require.NoError(t, err)
	content, err := os.ReadFile("docs/_licenses.md")
	require.NoError(t, err)
	snaps.MatchSnapshot(t, string(content))
	defer os.Remove("docs/_licenses.md")
}
