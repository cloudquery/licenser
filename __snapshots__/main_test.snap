
[TestReport - 1]
/-/-/-/
hub-title: Licenses
/-/-/-/

The following tools / packages are used in this plugin:

| Name | License |
|------|---------|
| github.com/spf13/cobra | Apache-2.0 |
| github.com/spf13/pflag | BSD-3-Clause |
| golang.org/x/sync/errgroup | BSD-3-Clause |
---
